### PR TITLE
Update

### PR DESCRIPTION
**Pull request by deemp**
Hi, @Arjan-Zuidema ! 

I updated the action a bit:
- Switched to TypeScript
- Added functionality to remove caches from multiple pages

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "",
   "main": "dist/index.js",
   "scripts": {
-    "build": "npx @vercel/ncc build src/index.js -m",
+    "build": "export NODE_OPTIONS=--openssl-legacy-provider && tsc && ncc build -o dist/ src/index.ts",
```
Comment (Arjan-Zuidema) [inline package.json:7]: @deemp  What is NODE_OPTIONS=--openssl-legacy-provider used for?

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    key?: string | undefined;
+    version?: string | undefined;
+    last_accessed_at?: string | undefined;
+    created_at?: string | undefined;
+    size_in_bytes?: number | undefined;
+  }
+
+  const results: Cache[] = []
+
+  for (let i = 1; i <= 100; i += 1) {
```
Comment (Arjan-Zuidema) [inline src/index.ts:23]: @deemp  Could you add a `break` when there are no more results? Now it always loops 100 times.

**Comment by Arjan-Zuidema** [comment]
Hi @deemp!

Thanks for your pull request! Looks good for the most part. I added some comments, if you can reply/fix those we can merge it 🙂

**Comment by deemp** [comment]
@Arjan-Zuidema, unfortunately, I don't see your comments

**Comment by Arjan-Zuidema** [comment]
Just scroll up in this window 🙂

**Comment by deemp** [comment]
Here's what I see :)
image

**Comment by Arjan-Zuidema** [comment]
Check files changed tab

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+jobs:
+  purge-old-caches:
```
Comment (deemp) [inline .github/workflows/ci.yaml]: Here's my comment

**Comment by deemp** [comment]
This is how comments should look like

**Review by Arjan-Zuidema — CHANGES_REQUESTED** [review]

**Comment by Arjan-Zuidema** [comment]
@deemp I added mentions and request changes

File: package.json (hunk 1 above)
Comment (deemp) [reply package.json:7]: I used nodejs from `nixpkgs`

```console
nix shell nixpkgs#nodejs_18
```

When I ran `npm run build`, I got

```console
ncc: Version 0.22.3
ncc: Compiling file index.js
Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:71:19)
    at Object.createHash (node:crypto:133:10)
    at hashOf (/home/eyjafjallajokull/Desktop/purge-cache/node_modules/@zeit/ncc/dist/ncc/index.js.cache.js:3:58216)
    at module.exports (/home/eyjafjallajokull/Desktop/purge-cache/node_modules/@zeit/ncc/dist/ncc/index.js.cache.js:3:60642)
    at runCmd (/home/eyjafjallajokull/Desktop/purge-cache/node_modules/@zeit/ncc/dist/ncc/cli.js.cache.js:1:47355)
    at 819 (/home/eyjafjallajokull/Desktop/purge-cache/node_modules/@zeit/ncc/dist/ncc/cli.js.cache.js:1:44227)
    at __webpack_require__ (/home/eyjafjallajokull/Desktop/purge-cache/node_modules/@zeit/ncc/dist/ncc/cli.js.cache.js:1:169)
    at startup (/home/eyjafjallajokull/Desktop/purge-cache/node_modules/@zeit/ncc/dist/ncc/cli.js.cache.js:1:339)
    at module.exports.8 (/home/eyjafjallajokull/Desktop/purge-cache/node_modules/@zeit/ncc/dist/ncc/cli.js.cache.js:1:371)
    at /home/eyjafjallajokull/Desktop/purge-cache/node_modules/@zeit/ncc/dist/ncc/cli.js.cache.js:1:381 {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}
```

I found the solution with the `NODE_OPTIONS` variable here

https://stackoverflow.com/questions/69394632/webpack-build-failing-with-err-ossl-evp-unsupported

File: src/index.ts (hunk 2 above)
Comment (deemp) [reply src/index.ts:23]: Sure

**Comment by Arjan-Zuidema** [comment]
@deemp What is NODE_OPTIONS=--openssl-legacy-provider used for?

**Comment by deemp** [comment]
@Arjan-Zuidema, I answered in a review. Is it visible?

image

**Comment by deemp** [comment]
My bad, I didn't submit

File: package.json [hunk 4]
```diff
@@ -22,5 +22,10 @@
   "dependencies": {
     "@actions/core": "^1.10.0",
     "@actions/github": "^5.1.1"
+  },
+  "devDependencies": {
+    "@zeit/ncc": "^0.22.3",
```
Comment (Arjan-Zuidema) [inline package.json]: Why did you change @vercel/ncc to @zeit/ncc? The package from @zeit has been deprecated in favor of @vercel

```
npm WARN deprecated @zeit/ncc@0.22.3: @zeit/ncc is no longer maintained. Please use @vercel/ncc instead.
```

File: package.json (hunk 1 above)
Comment (Arjan-Zuidema) [reply package.json:7]: Alright, seems fine to me

File: src/index.ts (hunk 2 above)
Comment (Arjan-Zuidema) [reply src/index.ts:23]: Thanks :)

File: package.json (hunk 4 above)
Comment (deemp) [reply package.json]: Because `@zeit/ncc` [worked](https://github.com/deemp/cache-nix-too/blob/eefa7c9691f7e13478dab17d58402e144d5bb07e/package.json#L37C7-L37C7) for me in another GH Action... and because I didn't notice you used `@vercel/ncc`. Switched to the latest `@vercel/ncc`

**Comment by deemp** [comment]
I updated the workflow to use it as an example

**Comment by deemp** [comment]
As there are a lot of commits, I guess it'd be better to squash them before merging

**Comment by deemp** [comment]
@Arjan-Zuidema , please, have a look

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+          path: |
+            ~/.npm
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json', 'package.json') }}-${{ steps.get-date.outputs.date }}
+          restore-keys: |
+            ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json', 'package.json') }}-
+      - name: Install packages and build the app
+        run: npm ci
+      - name: Commit & Push changes
+        run: |
+          git config --global user.name github-actions
```
Comment (Arjan-Zuidema) [inline .github/workflows/ci.yaml:44]: Can you use this for the config?

```
git config user.name "$GITHUB_ACTOR"
git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
```

**Review by Arjan-Zuidema — CHANGES_REQUESTED** [review]
Looks good! Just added one more thing, if you can fix that we can merge it :)

File: .github/workflows/ci.yaml (hunk 5 above)
Comment (deemp) [reply .github/workflows/ci.yaml:44]: done in https://github.com/MyAlbum/purge-cache/pull/2/commits/34a52afb283fc4f3fad284f6d4d47ddfcc4c485b

**Review by Arjan-Zuidema — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 5 above)
Comment (deemp) [reply .github/workflows/ci.yaml:44]: By the way, why did you request this change?
With the previous config, automated commits were done by `github-actions`.
With the new config, the commiter will be the one who triggers the workflow/has modified the workflow file (according to https://github.com/orgs/community/discussions/25067)

Comment (Arjan-Zuidema) [reply .github/workflows/ci.yaml:44]: We are using this config on our internal builds. But maybe it's nicer for public actions to use the way you config worked. You can revert it if you like 🙂

Comment (deemp) [reply .github/workflows/ci.yaml:44]: Ok :) Done in https://github.com/MyAlbum/purge-cache/pull/2/commits/eb51dbb33299946783edd2caba8a27415d757d2a

**Comment by deemp** [comment]
Workflows in my fork fail. Will look into it later

**Comment by deemp** [comment]
@Arjan-Zuidema , what can be the cause?
A cache can't be deleted.

https://github.com/deemp/purge-cache/actions/runs/5587233430/jobs/10212852447#step:3:23

**Comment by Arjan-Zuidema** [comment]
Seems like a permission issue. But I think that should not happen on a GitHub action run. I think this case needs to be handled so the rest of the run will continue. Could you add that in the PR? Just wrap it with a `try catch` and log the error

**Comment by Arjan-Zuidema** [comment]
This is merged in #3 🚀
